distribution/lib/Standard/Base/0.0.0-dev/src/Runtime/Lazy.enso

```diff
@@ -0,0 +1,75 @@
+import project.Any.Any
+import project.Error.Error
+import project.Nothing.Nothing
+import project.Panic.Caught_Panic
+import project.Panic.Panic
+import project.Runtime.Ref.Ref
+
+## Holds a value that is computed on first access.
```

Comment:
I believe there should be no `Lazy` type in the libraries, but the support shall come from the engine:
- #5085
- we have a prototype that shows how it all can be done
- the API will be simpler - just `~` in front of the atom field
- the access to the field will be transparent - no difference between lazy/cached/eager field
- the *performance will be better* if this is implemented in the engine
- we'll use less memory, if this is done in the engine as we eliminate delegation once the field is materialized/cached

Can we get a request from libraries to implement _"lazy atom fields"_?

Reply:
Ok, sounds all good!

I implemented this in pure Enso because I could do it quickly (it took about 1-2hr) and didn't want to wait for it to be implemented, as I wanted to be able to rely on this code straight away.

It also felt like it is simpler to do that way and at least for now we don't need this to be super performant - it's not used on some very hot paths - so I didn't want to take engine's time for something not critical.

But if we can get it replaced with a more efficient engine solution - I will be really happy to have it replaced. The benefits sound cool.

Reply:
However, I'd like to proceed with this PR without waiting for this implementation. So I would just keep the `Lazy` type for now. But once it is implemented in the engine, I'm more than happy to have it replaced with the "native" solution.

Reply:
Reported as
- #6134